sdk/core/core/internal/az_precondition_internal.h

```diff
@@ -43,23 +43,22 @@
 
 az_precondition_failed_fn az_precondition_failed_get_callback();
 
+AZ_INLINE void _az_precondition(bool condition)
+{
 #ifdef AZ_NO_PRECONDITION_CHECKING
-#define _az_PRECONDITION(condition)
+  (void)condition;
 #else
-#define _az_PRECONDITION(condition) \
-  do \
-  { \
-    if (!(condition)) \
-    { \
-      az_precondition_failed_get_callback()(); \
-    } \
-  } while (0)
+  if (!condition)
+  {
+    az_precondition_failed_get_callback()();
+  }
 #endif // AZ_NO_PRECONDITION_CHECKING
+}
 
-#define _az_PRECONDITION_RANGE(low, arg, max) _az_PRECONDITION((low <= arg && arg <= max))
+#define _az_PRECONDITION_RANGE(low, arg, max) _az_precondition((low <= arg && arg <= max))
```

Comment:
Can't this be an inline function too?

Reply:
I'll need to think about it, this is why I left it as a macro :) Maybe it needs to work for signed and unsigned ints of different sizes, and maybe even pointers. I'll take a look at usage. Maybe we'll remove this one at all (convert to conditions you pass to `_az_precondition()`)

Reply:
Jeff, I made `_az_precondition_range()` to take int32_t (other option would be to replace all `_az_PRECONDITION_RANGE()` usages with `_az_precondition(low <= arg && arg <= max))`

Reply:
Were all examples you found using int32_t?
No uses with other integer types?

Reply:
All but one. One is using int16_t.

Reply:
Great, then this int32_t version should work good enough then.

Reply:
Yes, I think it might be good enough for most cases. If there's any problem (using it with int64, for example), there should be warnings treated as errors. And for all the non-typical cases (pointers, unsigned, int64s) - they are not common, and can be written as `_az_precondition(min <= x && x <= max)`.